sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2513,7 +2514,8 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
   }
 
   override def visitDayTimeIntervalDataType(ctx: DayTimeIntervalDataTypeContext): DataType = {
-    DayTimeIntervalType
+    // TODO(SPARK-XXXXX): Support day-time interval fields in SQL
```

Comment:
Need to modify the parser rules. I plan to do that separately, and focus only on adding unit fields to the class `DayTimeIntervalType`.